controllers/front/OrderConfirmationController.php

```diff
@@ -45,31 +48,92 @@ public function init()
     {
         parent::init();
 
+        // If are coming to this page to finish free order, we need to run additional logic
         if (true === (bool) Tools::getValue('free_order')) {
             $this->checkFreeOrder();
         }
 
-        $this->id_cart = (int) (Tools::getValue('id_cart', 0));
-
-        $redirectLink = 'index.php?controller=history';
-
+        // We get the required data from URL
         $this->id_module = (int) (Tools::getValue('id_module', 0));
-        $this->id_order = Order::getIdByCartId((int) ($this->id_cart));
+        $this->id_order = (int) (Tools::getValue('id_order', 0));
         $this->secure_key = Tools::getValue('key', false);
-        $order = new Order((int) ($this->id_order));
+        $this->order = new Order((int) ($this->id_order));
 
+        // We run all the required checks and if something is smelly, we redirect him to order history
```

Comment:
One way to avoid having to write a comment is to use variable names
For example
```
        // We run all the required checks and if something is smelly, we redirect him to order history
```
can transform into
```
$somethingIsSmelly = (!Validate::isLoadedObject($this->order) || $this->secure_key != $this->order->secure_key);
if ($somethingIsSmelly)
```

Reply:
It's usually better to have dedicated method

```php
if (!$this->verifyOrder($this->order, $this->secure_key)) {
    // error
}
```